### PR TITLE
Have other rights appear in groupless

### PR DESCRIPTION
**Pull request by mnoack**

### CODE REVIEW

File: lib/right_on/right.rb [hunk 1]
```diff
@@ -34,6 +34,9 @@ def by_groups
         rights = []
         rights += regular_rights_with_group
         rights += restricted_rights_with_group
+        other_rights = Right.all - rights
+        other_rights.each { |right| right.group = 'Groupless' }
```
Comment (alvinypyim) [inline lib/right_on/right.rb]: Can we just append the groupless rights to the end of the resulting array instead of introducing a hardcoded group name?

Comment (mnoack) [inline lib/right_on/right.rb]: I think we can.

Comment (alvinypyim) [inline lib/right_on/right.rb]: Or maybe even clearer if we add a new method that just return the "groupless" rights.

File: lib/right_on/right.rb [hunk 2]
```diff
@@ -23,60 +23,66 @@ def associate_group(klass, group)
...
       def rights_yaml(file_path)
         @@rights_yaml = file_path
       end
 
       def by_groups
-        rights = []
-        rights += regular_rights_with_group
-        rights += restricted_rights_with_group
+        rights = regular_rights_with_group + restricted_rights_with_group
+        rights += (Right.all - rights)
```
Comment (alxberardi) [inline lib/right_on/right.rb]: Maybe I'm reading this wrong, but wouldn't this end up assigning all rights to the ``rights`` variable?
It reads like
```ruby
a = [1,2,3]
b = [4,5,6]
all = [1,2,3,4,5,6,7,8,9]
sum = a + b
sum = sum + all  - sum # => all
```
Or are the ``regular_rights_with_group`` and ``restricted_rights_with_group`` somehow decorated?

Comment (alxberardi) [inline lib/right_on/right.rb]: Ahh, I see the decoration happening below, with the assignment of the group. That answers the above question.

**Comment by alxberardi** [comment]
👍 (though this gem could use some more nice refactoring)

File: lib/right_on/right.rb (hunk 2 above)
Comment (vgmaster21) [inline lib/right_on/right.rb]: Optional suggestion. Maybe something like...
``
rights_with_groups = regular_rights_with_group + restricted_rights_with_group
other_rights = Rights.all - rights_with_groups
rights =  rights_with_groups + other_rights
``
It's an extra line but does make it easier to understand the logic.

Comment (mnoack) [inline lib/right_on/right.rb]: @alxberardi Shall I make it like the above and merge?

Comment (alxberardi) [inline lib/right_on/right.rb]: I'm happy how it is now. If you want to implement @vgmaster21 's suggestion feel free to. I just didn't immediately notice the assignment of the groups to the rights.

**Pull request merged**